ods_ci/tests/Tests/650__distributed_workloads/test-run-kueue-e2e-tests.robot

```diff
@@ -0,0 +1,77 @@
+*** Settings ***
+Documentation     Kueue E2E tests - https://github.com/kubernetes-sigs/kueue.git
+Suite Setup       Prepare Kueue E2E Test Suite
+Suite Teardown    Teardown Kueue E2E Test Suite
+Library           OperatingSystem
+Library           Process
+Library           OpenShiftLibrary
+
+
+*** Variables ***
+${KUEUE_DIR}            kueue
+${KUEUE_REPO_URL}       %{KUEUE_REPO_URL=https://github.com/kubernetes-sigs/kueue.git}
+${KUEUE_REPO_BRANCH}    %{KUEUE_REPO_BRANCH=main}
+${WORKER_NODE}          ${EMPTY}
+
+
+*** Test Cases ***
+Run E2E test
+    [Documentation]    Run ginkgo E2E single cluster test
+    [Tags]  Kueue
+    ...     DistributedWorkloads
+    Run Kueue E2E Test    e2e_test.go
+
+Run Visibility test
+    [Documentation]    Run ginkgo visibilty single cluster test
+    [Tags]  Kueue
+    ...     DistributedWorkloads
+
+    Enable Visibility Feature Gate
+    Run Kueue E2E Test    visibility_test.go
+
+
+*** Keywords ***
+Prepare Kueue E2E Test Suite
+    [Documentation]    Prepare Kueue E2E Test Suite
+    ${result} =    Run Process    git clone -b ${KUEUE_REPO_BRANCH} ${KUEUE_REPO_URL} ${KUEUE_DIR}
+    ...    shell=true    stderr=STDOUT
+    Log To Console    ${result.stdout}
+    IF    ${result.rc} != 0
+        FAIL    Unable to clone kueue repo ${KUEUE_REPO_URL}:${KUEUE_REPO_BRANCH}:${KUEUE_DIR}
+    END
+
+    Log To Console    Install the latest development version of kueue ...
+    ${return_code}    ${output}    Run And Return Rc And Output    kubectl apply --server-side -k "github.com/kubernetes-sigs/kueue/config/default?ref=main"
+    Log To Console    ${output}
+    Should Be Equal As Integers   ${return_code}   0  msg=Error detected while installing kueue
+
+    # Add label instance-type=on-demand on worker node
+    ${result} =    Run Process    oc get nodes -o name | tail -n1
+    ...    shell=true    stderr=STDOUT
+    Set Suite Variable    ${WORKER_NODE}    ${result.stdout}
+    ${return_code} =    Run And Return Rc    oc label ${WORKER_NODE} instance-type=on-demand
+    Should Be Equal As Integers  ${return_code}   0   msg=Fail to label worker node with instance-type=on-demand
+
+Teardown Kueue E2E Test Suite
+    [Documentation]    Teardown Kueue E2E Test Suite
+    Log To Console    Uninstall Kueue ...
+    ${return_code}    ${output}    Run And Return Rc And Output    kubectl delete -k "github.com/kubernetes-sigs/kueue/config/default?ref=main"
+    Log To Console    ${output}
+    Should Be Equal As Integers   ${return_code}   0  msg=Error detected while uninstalling kueue
+
+    # Remove label instance-type=on-demand from worker node
+    ${return_code} =    Run And Return Rc    oc label ${WORKER_NODE} instance-type-
```

Comment:
Does this code really remove label from node?

Reply:
Yes, i have tested it

Reply:
Thanks